packages/analytics-js-plugins/src/ketchConsentManager/index.ts

```diff
@@ -69,20 +69,44 @@ const KetchConsentManager = (): ExtensionPlugin => ({
       const allowedConsentIds = state.consents.data.value.allowedConsentIds as string[];
 
       try {
-        const { ketchConsentPurposes } = destConfig;
+        const { ketchConsentPurposes, consentManagement } = destConfig;
+        const matchPredicate = (consent: string) => allowedConsentIds.includes(consent);
 
-        // If the destination do not have this mapping events will be sent.
-        if (!ketchConsentPurposes || ketchConsentPurposes.length === 0) {
-          return true;
-        }
+        // Generic consent management
+        if (consentManagement) {
+          // Get the corresponding consents for the destination
+          const cmpConsents = consentManagement.find(
+            c => c.provider === state.consents.provider.value,
+          )?.consents;
+
+          // If there are no consents configured for the destination for the current provider, events should be sent.
+          if (!cmpConsents) {
+            return true;
+          }
+
+          const configuredConsents = cmpConsents.map(c => c.consent.trim()).filter(n => n);
+
+          // match the configured consents with user provided consents as per
+          // the configured resolution strategy
+          switch (state.consents.resolutionStrategy.value) {
+            case 'or':
+              return configuredConsents.some(matchPredicate) || configuredConsents.length === 0;
+            case 'and':
+            default:
+              return configuredConsents.every(matchPredicate);
+          }
 
-        const purposes = ketchConsentPurposes.map(p => p.purpose).filter(n => n);
+          // Legacy cookie consent management
+          // To be removed once the source config API is updated to support generic consent management
```

Comment:
add TODO: in this comment in order to locate future tasks easily

Reply:
Done.